src/net/sourceforge/kolmafia/persistence/DebugDatabase.java

```diff
@@ -2264,7 +2268,7 @@ private static void loadScrapeData(final StringArray array, final String fileNam
   }
 
   private static void saveScrapeData(
-      final Iterator<Integer> it, final StringArray array, final String fileName) {
+      final Iterator<Integer> it, final Map<Integer, String> stringMap, final String fileName) {
```

Comment:
And this method right here is the demonstration that you don't need a LInkedHashMap. It does not iterate over the map. It passed in an Iterator over the IDs of a data set - items, effects, outfits, whatever - and fetches  using those indices.

You could pass in an ArrayList<String> or a HashMap<String> and it would generate the output file in exactly the same order, whether or not there are gaps in the data.